src/rpc.rs

```diff
@@ -0,0 +1,127 @@
+/********************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+use crate::UPClientMqtt;
+use async_std::future::timeout;
+use async_trait::async_trait;
+use std::{sync::Arc, time::Duration};
+use tokio::sync::mpsc;
+use up_rust::{
+    RpcClient, RpcClientResult, UAttributesValidators, UListener, UMessage, UMessageError, UStatus,
+    UUri,
+};
+
+struct RpcListener {
+    sender: mpsc::Sender<RpcClientResult>,
+    correlation_id: String,
+}
+
+impl RpcListener {
+    pub fn new(sender: mpsc::Sender<RpcClientResult>, correlation_id: String) -> Self {
+        Self {
+            sender,
+            correlation_id,
+        }
+    }
+}
+
+#[async_trait]
+impl UListener for RpcListener {
+    // TODO: Handle send errors
```

Comment:
:+1: Yup

Reply:
While Rpc Client was originally intended to go in with this PR I backed out those changes as I know there is a restructure of how the RpcClient is going to be handled and where it will live. RpcClient will be a separate PR if it is meant to go into this up-transport repo